server/main.go

```diff
@@ -0,0 +1,278 @@
+package main
+
+import (
+	"math/big"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gochain-io/explorer/server/backend"
+	"github.com/gochain-io/explorer/server/models"
+
+	"github.com/gochain-io/gochain/ethclient"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/urfave/cli"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-chi/cors"
+)
+
+var ethClient *ethclient.Client
+var backendInstance *backend.Backend
+var wwwRoot string
+
+func parseSkipLimit(r *http.Request) (int, int) {
+	limitS := r.URL.Query().Get("limit")
+	skipS := r.URL.Query().Get("skip")
+	skip := 0
+	limit := 100
+	if skipS != "" {
+		skip, _ = strconv.Atoi(skipS)
+	}
+	if limitS != "" {
+		limit, _ = strconv.Atoi(limitS)
+	}
+
+	if skip < 0 {
+		skip = 0
+	}
+
+	if limit > 500 || limit <= 0 {
+		limit = 100
+	}
```

Comment:
Wouldn't setting `> 500` to `500` make more sense?

But why cap it at all? This prevents fetching an atomic snapshot of the data. I think enforcing this cap on the *front end* makes sense, so that the browser isn't overloaded, but maybe not on the backend.